### ISSUE TITLE
Function names are dropped from async functions with asyncToGenerator

### ISSUE THREAD

**Issue by ide** [babel/babel#2481]
When you define a named async function and enable the `asyncToGenerator` option, the name of the function is dropped. This is with 5.8.23 (babel-core 5.8.25). This input:

```js
async function x() { }
```

Produces:
```js
var x = _asyncToGenerator(function* () {});
```

Instead of:
```js
var x = _asyncToGenerator(function* x() {});
```
---
.babelrc looks like this:
```json
{
  "stage": 0,
  "optional": [
    "asyncToGenerator"
  ]
}

```

**Comment by zloirock** [comment]
#1618 related.

**Comment by sebmck** [comment]
Closed via 9320d661f911ee598315554ee807c817f7a45900.
